pkg/manager/manager.go

```diff
@@ -458,25 +458,18 @@ func (m *Manager) DestroyDaemon(d *daemon.Daemon) error {
 // 1. Don't erase ever written record
 // 2. Just recover nydusd daemon states to manager's memory part.
 // 3. Manager in SharedDaemon mode should starts a nydusd when recovering
-func (m *Manager) Recover(ctx context.Context) (map[string]*daemon.Daemon, map[string]*daemon.Daemon, error) {
-	// Collected deserialized daemons that need to be recovered.
-	recoveringDaemons := make(map[string]*daemon.Daemon, 0)
-	liveDaemons := make(map[string]*daemon.Daemon, 0)
-
+func (m *Manager) Recover(ctx context.Context,
+	recoveringDaemons *map[string]*daemon.Daemon, liveDaemons *map[string]*daemon.Daemon) error {
 	if err := m.store.WalkDaemons(ctx, func(s *daemon.States) error {
-		log.L.Debugf("found daemon states %#v", s)
+		if s.FsDriver != m.FsDriver {
+			return nil
```

Comment:
Maybe give a warning here.

Reply:
Now we are going to support multiple daemon mode, so it's legal now:)